server/docker-compose.yml

```diff
@@ -0,0 +1,112 @@
+name: bymr
+
+volumes:
+  db:
+  pgadmin:
+
+networks:
+  bymr:
+
+services:
+  # Database initialization service (runs once)
+  db-init:
+    build: .
+    environment:
+      - DB_NAME=${DB_NAME}
+      - DB_HOST=db
+      - DB_PORT=5432
+      - DB_USER=${DB_USER}
+      - DB_PASSWORD=${DB_PASSWORD}
+    networks:
+      - bymr
+    depends_on:
+      db:
+        condition: service_healthy
+    command: npm run db:init
+    restart: "no"
+
+  # API server
+  web:
+    build: .
+    container_name: bymr-api-server
+    environment:
+      - ENV=${ENV}
+      - BASE_URL=${BASE_URL}
+      - PORT=3001
+      - DB_NAME=${DB_NAME}
+      - DB_HOST=db
+      - DB_PORT=5432
+      - DB_USER=${DB_USER}
+      - DB_PASSWORD=${DB_PASSWORD}
+      - SECRET_KEY=${SECRET_KEY}
+      - REDIS_URL=redis://redis:6379
+    networks:
+      - bymr
+    ports:
+      - ${API_PORT}:3001
+    develop:
+      watch:
+        - action: sync
+          path: server
+          target: /app
+    depends_on:
+      redis:
+        condition: service_healthy
+      db:
+        condition: service_healthy
+      db-init:
+        condition: service_completed_successfully
+
+  # Database
+  db:
+    image: postgres:16
+    container_name: bymr-database
+    environment:
+      - POSTGRES_DB=${DB_NAME}
+      - POSTGRES_USER=${DB_USER}
+      - POSTGRES_PASSWORD=${DB_PASSWORD}
+    ports:
+      - ${DB_PORT}:5432
```

Comment:
Maybe introduce a default? Like this:
`${DB_PORT:-5432}:5432` 

You also specified a default value for PG Admin

Reply:
A more sensible way would be to not expose the port on production, all services are connected to each other internally. Exposing does make them available on the host.

In prod, never expose the port directly